base_user_role/models/role.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 ABF OSIELL <http://osiell.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import datetime
+import logging
+
+from openerp import api, fields, models
+
+
+_logger = logging.getLogger(__name__)
+
+
+class ResUsersRole(models.Model):
+    _name = 'res.users.role'
+    _inherits = {'res.groups': 'group_id'}
+    _description = "User role"
+
+    group_id = fields.Many2one(
+        'res.groups', required=True, ondelete='cascade',
+        readonly=True, string=u"Associated group")
+    line_ids = fields.One2many(
+        'res.users.role.line', 'role_id', string=u"Users")
+    user_ids = fields.One2many(
+        'res.users', string=u"Users", compute='_compute_user_ids')
+
+    @api.multi
+    @api.depends('line_ids')
+    def _compute_user_ids(self):
+        for role in self:
+            role.user_ids = role.line_ids.mapped('user_id')
+
+    @api.model
+    def create(self, vals):
+        new_record = super(ResUsersRole, self).create(vals)
+        new_record.update_users()
+        return new_record
+
+    @api.multi
+    def write(self, vals):
+        res = super(ResUsersRole, self).write(vals)
+        self.update_users()
+        return res
+
+    @api.multi
+    def update_users(self):
+        """Update all the users concerned by the roles identified by `ids`."""
+        users = self.mapped('user_ids')
+        if users:
+            users.set_groups_from_roles()
+        return True
+
+    @api.model
+    def cron_update_users(self):
+        logging.info(u"Update user roles")
+        self.search([]).update_users()
+
+
+class ResUsersRoleLine(models.Model):
+    _name = 'res.users.role.line'
+    _description = 'Users associated to a role'
+
+    role_id = fields.Many2one(
+        'res.users.role', string=u"Role")
+    user_id = fields.Many2one(
+        'res.users', string=u"User")
+    date_from = fields.Date(u"From")
+    date_to = fields.Date(u"To")
+    is_active = fields.Boolean(u"Active", compute='_compute_is_active')
```

Comment:
Indeed, but I was unable to set a working `domain` on the XML views to display all the lines (as we don't want to hide them from the administrator in any case, we just want to filter them when setting groups). If I set the domain on the Python field, it works fine, but then it does not dispense to filter the lines in the `ResUsers.set_groups_from_roles` method. And as I've seen no benefit to use the standard `active` field I ended up adding my own.